app/scripts/lib/transaction/smart-transactions.ts

```diff
@@ -313,7 +313,9 @@ class SmartTransactionHook {
       signedTransactions,
       signedCanceledTransactions: [],
       txParams: this.#txParams,
-      transactionMeta: this.#transactionMeta,
+      // TODO: Replace `any` with type - version mismatch between smart-transactions-controller and transaction-controller breaking type safety
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
Nit: `@ts-expect-error` is preferred because it will flag when the override is no longer needed

See https://github.com/MetaMask/contributor-docs/blob/main/docs/typescript.md#if-accompanied-by-a-todo-comment-ts-expect-error-is-acceptable-to-use-for-marking-errors-that-have-clear-plans-of-being-resolved for more details

```suggestion
      // @ts-expect-error version mismatch between smart-transactions-controller and transaction-
      // controller breaking type safety
```